spring-cloud-gcp-bigquery/src/main/java/com/google/cloud/spring/bigquery/core/BigQueryTemplate.java

```diff
@@ -96,12 +98,15 @@ public class BigQueryTemplate implements BigQueryOperations {
    * @param bqInitSettings Properties required for initialisation of this class
    * @param taskScheduler the {@link TaskScheduler} used to poll for the status of long-running
    *     BigQuery operations
+   * @param jsonWriterExecutorService the {@link ExecutorService} used to run the thread required
+   *     for the BigQuery JSON Writer
    */
   public BigQueryTemplate(
       BigQuery bigQuery,
       BigQueryWriteClient bigQueryWriteClient,
       Map<String, Object> bqInitSettings,
-      TaskScheduler taskScheduler) {
+      TaskScheduler taskScheduler,
+      ExecutorService jsonWriterExecutorService) {
```

Comment:
This is a breaking change. Either add a setter or an overloaded constructor.

Reply:
My bad, I thought that we have updated the `bigQueryTemplate` bean @ `GcpBigQueryAutoConfiguration` , so we are safe from this breaking change! But you are right and a user might directly initialise `BigQueryTemplate`.

So, I have added  a overloaded constructor where the user need not pass `jsonWriterExecutorService `, added another setter so that they can override `jsonWriterThreadPoolSize`